tests/test_tasks/test_task.py

```diff
@@ -1,77 +1,42 @@
-        retval = openml.tasks.get_task(1)
-        self.assertEqual(patch.call_count, 1)
-        self.assertIsInstance(retval, openml.OpenMLTask)
-        self.assertEqual(retval.class_labels, 'LA')
-
-    def test_get_X_and_Y(self):
-        # Classification task
-        task = openml.tasks.get_task(1)
-        X, Y = task.get_X_and_y()
-        self.assertEqual((898, 38), X.shape)
-        self.assertIsInstance(X, np.ndarray)
-        self.assertEqual((898, ), Y.shape)
-        self.assertIsInstance(Y, np.ndarray)
-        self.assertEqual(Y.dtype, int)
-
-        # Regression task
-        task = openml.tasks.get_task(631)
-        X, Y = task.get_X_and_y()
-        self.assertEqual((52, 2), X.shape)
-        self.assertIsInstance(X, np.ndarray)
-        self.assertEqual((52,), Y.shape)
-        self.assertIsInstance(Y, np.ndarray)
-        self.assertEqual(Y.dtype, float)
-
-    def test_tagging(self):
-        task = openml.tasks.get_task(1)
-        tag = "testing_tag_{}_{}".format(self.id(), time())
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 0)
-        task.push_tag(tag)
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 1)
-        self.assertIn(1, task_list)
-        task.remove_tag(tag)
-        task_list = openml.tasks.list_tasks(tag=tag)
-        self.assertEqual(len(task_list), 0)
-
-    def test_get_train_and_test_split_indices(self):
-        openml.config.cache_directory = self.static_cache_dir
-        task = openml.tasks.get_task(1882)
-        train_indices, test_indices = task.get_train_test_split_indices(0, 0)
-        self.assertEqual(16, train_indices[0])
-        self.assertEqual(395, train_indices[-1])
-        self.assertEqual(412, test_indices[0])
-        self.assertEqual(364, test_indices[-1])
-        train_indices, test_indices = task.get_train_test_split_indices(2, 2)
-        self.assertEqual(237, train_indices[0])
-        self.assertEqual(681, train_indices[-1])
-        self.assertEqual(583, test_indices[0])
-        self.assertEqual(24, test_indices[-1])
-        self.assertRaisesRegexp(ValueError, "Fold 10 not known",
-                                task.get_train_test_split_indices, 10, 0)
-        self.assertRaisesRegexp(ValueError, "Repeat 10 not known",
-                                task.get_train_test_split_indices, 0, 10)
+    # task id will be set from the
+    # extending classes
+
+    def setUp(self):
+
+        super(OpenMLTaskTest, self).setUp()
+        self.task_id = 11
+        self.estimation_procedure = 23
+
+    @classmethod
+    def setUpClass(cls):
+        if cls is OpenMLTaskTest:
+            raise unittest.SkipTest(
+                "Skip OpenMLTaskTest tests,"
+                " it's a base class"
+            )
+        super(OpenMLTaskTest, cls).setUpClass()
+
+    def test_download_task(self):
+
+        openml.tasks.get_task(self.task_id)
+
+    def test_upload_task(self):
```

Comment:
Would it be possible to create a task which will never be on the server so that the upload can definitely be tested? For example by uploading a dataset prior to creating a task?

Reply:
This is already done, the tasks in the unit tests were using different evalutation strategies which were not present in the test server and the upload was successful.

Reply:
Yes, but they'll be there in the next iteration, right?

Reply:
Yes, normally this would be easily doable if the remove_task functionality would be on the connector. However it is only available in the website. 
Are you suggesting to upload a dataset continuously so that the id always changes resulting in a new task?
Is there no plan for the moment to port the remove_task to the connector?

Reply:
I currently don't plan to add this, but I think @janvanrijn added the delete functionality in PR #620. However, I wouldn't mind if this was kept on the server for now.

Reply:
We should actually address this in #261 and actually also remove datasets, flows and runs (at some point, not now).